scheduler/scheduler.go

```diff
@@ -187,6 +187,27 @@ func (s *Scheduler) Schedule(ctx context.Context) error {
 
 	hostPlanningStart := time.Now()
 
+	catcher := grip.NewCatcher()
+	existing, err := host.Count(host.IsUninitialized)
+	catcher.Add(err)
+	running, err := host.Count(host.IsRunning)
+	catcher.Add(err)
+
+	if catcher.HasErrors() {
+		return errors.WithStack(catcher.Resolve())
+	}
+
+	grip.Notice(message.Fields{
+		"runner":    RunnerName,
+		"operation": "removing stale intent hosts",
+		"existing":  existing,
+		"running":   running,
+	})
+
+	if err := host.RemoveAllInitializing(); err != nil {
```

Comment:
Would this make more sense to do in the monitor? Doesn't make too much of a difference but since it's the one trying to terminate hosts anyway and runs less frequently, it may be better there

Reply:
Needs to happen right before the scheduler runs so you don't allocate hosts based on the previous scheduler run's hosts?